### PR TITLE
feat: add status

### PR DESCRIPTION
**Pull request by daybrush**

### CODE REVIEW

File: src/InfiniteGrid.ts [hunk 1]
```diff
@@ -6,28 +6,36 @@ import {
...
   OnContentError,
+  ItemRenderer,
 } from "@egjs/grid";
-import { EVENTS } from "./consts";
+import { EVENTS, ITEM_TYPE, STATUS_TYPE } from "./consts";
 import { GroupManager } from "./GroupManager";
-import { Infinite, OnInfiniteChange, OnInfiniteRequestAppend, OnInfiniteRequestPrepend } from "./Infinite";
-import { InfiniteGridItem } from "./InfiniteGridItem";
+import {
+  Infinite, OnInfiniteChange } from "./Infinite";
```
Comment (WoodNeck) [inline src/InfiniteGrid.ts:15]: Unnecessary newline here

File: src/InfiniteGrid.ts [hunk 2]
```diff
@@ -6,28 +6,36 @@ import {
...
+  InfiniteGridGetterSetter, toArray, convertInsertedItems, findIndex,
+  findLastIndex, isString,
+} from "./utils";
 
 
 /**
- * A module used to arrange card elements including content infinitely according to layout type. With this module, you can implement various layouts composed of different card elements whose sizes vary. It guarantees performance by maintaining the number of DOMs the module is handling under any circumstance
- * @ko 콘텐츠가 있는 카드 엘리먼트를 레이아웃 타입에 따라 무한으로 배치하는 모듈. 다양한 크기의 카드 엘리먼트를 다양한 레이아웃으로 배치할 수 있다. 카드 엘리먼트의 개수가 계속 늘어나도 모듈이 처리하는 DOM의 개수를 일정하게 유지해 최적의 성능을 보장한다
+ * A module used to arrange items including content infinitely according to layout type. With this module, you can implement various layouts composed of different items whose sizes vary. It guarantees performance by maintaining the number of DOMs the module is handling under any circumstance
+ * @ko 콘텐츠가 있는 아이템를 레이아웃 타입에 따라 무한으로 배치하는 모듈. 다양한 크기의 아이템를 다양한 레이아웃으로 배치할 수 있다. 아이템의 개수가 계속 늘어나도 모듈이 처리하는 DOM의 개수를 일정하게 유지해 최적의 성능을 보장한다
```
Comment (WoodNeck) [inline src/InfiniteGrid.ts:38]: You replaced as "아이템를"

File: src/InfiniteGrid.ts [hunk 3]
```diff
@@ -447,46 +587,120 @@ class InfiniteGrid<Options extends InfiniteGridOptions = InfiniteGridOptions> ex
...
-  private _onRequestPrepend = (e: OnInfiniteRequestPrepend): void => {
+  private _onRequestPrepend = (e: OnRequestInsert): void => {
     // TODO LOADING
     this.trigger(EVENTS.REQUEST_PREPEND, {
       groupKey: e.groupKey,
+      nextGroupKey: e.nextGroupKey,
     });
   }
 
+  private _checkVirtualGroups() {
```
Comment (WoodNeck) [inline src/InfiniteGrid.ts:650]: Can you add some documentation about the return value of this method?
Like, it does check the virtual groups and return `boolean` about what?

File: src/types.ts [hunk 4]
```diff
@@ -45,35 +62,44 @@ export interface InfiniteGridOptions extends GridOptions {
  * @typedef
  * @memberof InfiniteGrid
  * @property - Last group key. <ko>마지막 그룹의 키.</ko>
+ * @property - The key of the next group that should replace the placeholder. <ko>placeholder를 대체해야 할 다음 그룹의 키.</ko>
  */
 export interface OnRequestAppend {
   groupKey: string | number | undefined;
+  nextGroupKey: string | number | undefined;
```
Comment (WoodNeck) [inline src/types.ts:69]: How about using `?` here, like the others?

**Review by WoodNeck — APPROVED** [review]
LGTM

File: src/utils.ts [hunk 5]
```diff
@@ -206,3 +223,54 @@ export function findLastIndex<T>(arr: T[], callback: (value: T, index: number) =
...
+}
+
+export function range(length: number): number[] {
+  const arr: number[] = [];
+  for (let i = 0; i < length; ++i) {
+    arr.push(i);
+  }
+  return arr;
+}
+
```
Comment (mixed) [inline src/utils.ts:261]: How about making it simpler?
```js
[...Array(10).keys()]
```

**Review by mixed — APPROVED** [review]
LGTM

File: src/utils.ts (hunk 5 above)
Comment (daybrush) [reply src/utils.ts:261]: @mixed 

keys is not supported in ie.

**Pull request merged**